core/storage/beam_job/gae_models.py

```diff
@@ -0,0 +1,169 @@
+    be unique:
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#resource:-job
+    """
+
+    # The name of the job class that implements the job's logic.
+    job_name = datastore_services.StringProperty(indexed=True, required=True)
+    # The arguments provided to the job run.
+    job_arguments = (
+        datastore_services.StringProperty(indexed=True, repeated=True))
+    # The state of the job at the time the model was last updated.
+    latest_job_state = datastore_services.StringProperty(indexed=True, choices=[
+        BeamJobState.RUNNING.value,
+        BeamJobState.PENDING.value,
+        BeamJobState.STOPPED.value,
+        BeamJobState.CANCELLING.value,
+        BeamJobState.CANCELLED.value,
+        BeamJobState.DRAINING.value,
+        BeamJobState.DRAINED.value,
+        BeamJobState.UPDATED.value,
+        BeamJobState.DONE.value,
+        BeamJobState.FAILED.value,
+        BeamJobState.UNKNOWN.value,
+    ], required=True)
+
+    @property
+    def in_terminal_state(self):
+        """Returns whether the job run has reached a terminal state and is no
+        longer running.
+
+        Returns:
+            bool. Whether the job has reached a terminal state.
+        """
+        return self.latest_job_state in [BeamJobState.CANCELLED.value,
+                                         BeamJobState.DRAINED.value,
+                                         BeamJobState.UPDATED.value,
+                                         BeamJobState.DONE.value,
+                                         BeamJobState.FAILED.value]
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model doesn't contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(BeamJobRunModel, cls).get_export_policy(), **{
+            'job_name': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'job_arguments': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'latest_job_state': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+
+class BeamJobRunResultModel(base_models.BaseModel):
+    """Represents the result of an Apache Beam job.
+
+    IDs are generated by Google Cloud Dataflow, not Oppia, and are guaranteed to
+    be unique:
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#resource:-job
+    """
+
+    # The standard text output generated by the corresponding Apache Beam job.
+    stdout = datastore_services.TextProperty(indexed=False)
+    # The error output generated by the corresponding Apache Beam job.
+    stderr = datastore_services.TextProperty(indexed=False)
```

Comment:
Do we need these fields to be required=True? Also, the default value for indexed is False, right? (not sure!)

Reply:
No they won't be, some jobs (like validation jobs) will only ever report errors and others will only report information, so this is fine.